lab_5_scrapper/scrapper.py

```diff
@@ -120,46 +254,96 @@ def __init__(self, full_url: str, article_id: int, config: Config) -> None:
+        self.article_id = article_id
+        self.config = config
+        self.article = Article(url=full_url, article_id=article_id)
 
     def _fill_article_with_text(self, article_soup: BeautifulSoup) -> None:
         """
         Finds text of article
         """
-        pass
+        texts_bs = article_soup.find('div', {
+            'class':
+                'elementor-element elementor-element-6732342 elementor-widget elementor-widget-theme-post-content'})
+        texts = texts_bs.find_all('p')
+        result = ''
+        for el in texts:
+            txt = el.text
+            txt = txt.replace('\n\t\t\t\t', '')
+            txt = txt.replace('\t\t\t', '')
+            if txt != '\xa0':
+                result = ' '.join([result, txt])
+        self.article.text = result
 
     def _fill_article_with_meta_information(self, article_soup: BeautifulSoup) -> None:
         """
         Finds meta information of article
         """
-        pass
+        title_bs = article_soup.find('h1', {'class': 'elementor-heading-title elementor-size-default'})
+        self.article.title = title_bs.text
+
+        date_bs = article_soup.find('span',
+                                    {'class': 'elementor-icon-list-text elementor-post-info__item elementor-post-info__item--type-date'})
+        date_txt = re.search(r'\d{2}/\d{2}/\d{4}', date_bs.text)
+        self.article.date = date_txt[0]
+
+        auth_bs = article_soup.find('span', {
+            'class': 'elementor-icon-list-text elementor-post-info__item elementor-post-info__item--type-author'})
+        auth_txt = re.search(r'\w+\s\w+', auth_bs.text)
+        self.article.author = auth_txt[0]
+
+        topic_bs = article_soup.find('h3', {'class': 'elementor-heading-title elementor-size-default'})
+        topic_txt = topic_bs.find('span').text
+        self.article.topics = topic_txt
+
+        self.article.article_id = self.article_id
+        self.article.url = self.full_url
 
     def unify_date_format(self, date_str: str) -> datetime.datetime:
         """
         Unifies date format
         """
-        pass
+        return datetime.datetime.strptime(date_str, '%d/%m/%y')
 
     def parse(self) -> Union[Article, bool, list]:
         """
         Parses each article
         """
-        pass
+        response = make_request(self.full_url, self.config)
+        article_bs = BeautifulSoup(response.text, 'lxml')
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        return self.article
+
 
 
 def prepare_environment(base_path: Union[Path, str]) -> None:
     """
     Creates ASSETS_PATH folder if no created and removes existing folder
     """
-    pass
+
+    if base_path.exists():
+        shutil.rmtree(base_path)
+
+    base_path.mkdir(parents=True)
+
 
 
 def main() -> None:
     """
     Entrypoint for scrapper module
     """
-    pass
-
+    base_path = Path('C:\\pyproj\\hse\\2022-2-level-ctlr\\tmp\\articles')
```

Comment:
use ASSETS_PATH from constants, otherwise it never works on CI side

Reply:
Done